src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,241 @@
+
+        public override int KeySize
+        {
+            get { return _impl.KeySize; }
+            set { _impl.KeySize = value; }
+        }
+
+        public override KeySizes[] LegalKeySizes => 
+            _impl.LegalKeySizes; // Csp Windows and RSAOpenSsl are the same (384, 16384, 8)
+
+        // PersistKeyInCsp has no effect in Unix
+        public bool PersistKeyInCsp { get; set; }
+
+        public bool PublicOnly
+        {
+            get { return _publicOnly; }
+        }
+
+        public override string SignatureAlgorithm => _impl.SignatureAlgorithm;
+
+        public override byte[] SignData(Stream data, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) =>
+            _impl.SignData(data, hashAlgorithm, padding);
+
+        public override byte[] SignData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) =>
+            _impl.SignData(data, offset, count, hashAlgorithm, padding);
+
+        public byte[] SignData(byte[] buffer, int offset, int count, object halg) =>
+            _impl.SignData(buffer, offset, count, HashAlgorithmNames.ObjToHashAlgorithmName(halg), RSASignaturePadding.Pkcs1);
+
+        public byte[] SignData(byte[] buffer, object halg) =>
+            _impl.SignData(buffer, HashAlgorithmNames.ObjToHashAlgorithmName(halg), RSASignaturePadding.Pkcs1);
+
+        public byte[] SignData(Stream inputStream, object halg) =>
+            _impl.SignData(inputStream, HashAlgorithmNames.ObjToHashAlgorithmName(halg), RSASignaturePadding.Pkcs1);
+
+        public override byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) =>
+            _impl.SignHash(hash, hashAlgorithm, padding);
+
+        public byte[] SignHash(byte[] rgbHash, string str)
+        {
+            if (rgbHash == null)
+                throw new ArgumentNullException(nameof(rgbHash));
+
+            HashAlgorithmName algName = HashAlgorithmNames.NameOrOidToHashAlgorithmName(str);
+            return _impl.SignHash(rgbHash, algName, RSASignaturePadding.Pkcs1);
+        }
+
+        public override string ToXmlString(bool includePrivateParameters) => _impl.ToXmlString(includePrivateParameters);
+
+        public bool VerifyData(byte[] buffer, object halg, byte[] signature) =>
+            _impl.VerifyData(buffer, signature, HashAlgorithmNames.ObjToHashAlgorithmName(halg), RSASignaturePadding.Pkcs1);
+
+        public override bool VerifyData(byte[] data, int offset, int count, byte[] signature, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) =>
+            _impl.VerifyData(data, offset, count, signature, hashAlgorithm, padding);
+
+        public override bool VerifyHash(byte[] hash, byte[] signature, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
+        {
+            if (signature == null)
+                throw new ArgumentNullException(nameof(signature));
+
+            // _impl does remaining parameter validation
```

Comment:
_impl could support PSS.  This doesn't.  More accurately, this only supports PKCS1 padding.

(Currently neither macOS or Linux/OpenSSL RSA have PSS, but that could change, and this shouldn't)